driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1857,6 +1833,8 @@ public void onFailure(Throwable t) {
                     s.updateCreatedPools(blockingExecutor);
 
             } finally {
+                if (preExistentConnection != null && !preExistentConnection.hasPool())
```

Comment:
I am including this statement in this recently-added try-finally block; the original code had a try-finally block that spanned wider, starting from the `if(isClosed())` statement.